### PR TITLE
feat: Improve global `cmd+k` menu

### PR DESCRIPTION
**Pull request by BrendanChou**
OVERVIEW:
- Add localization
- Add page navigation
- Move specific preferences (e.g. themeing) into Preferences/Display
Screenshot 2024-07-09 at 14 09 11

### CODE REVIEW

File: src/views/menus/useGlobalCommands.tsx [hunk 1]
```diff
@@ -42,83 +46,66 @@ export const useGlobalCommands = (): MenuConfig<string, string> => {
...
+        ...(showVaults
+          ? [
+              {
+                value: 'vaults',
+                slotBefore: <Icon iconName={IconName.Governance} />,
+                label: stringGetter({ key: STRING_KEYS.VAULTS }),
+                onSelect: () => navigate(AppRoute.Vaults),
+              },
+            ]
+          : []),
```
Comment (BrendanChou) [inline src/views/menus/useGlobalCommands.tsx:85]: I saw this being done is a different way elsewhere (search for `showVaults`) but I couldn't get the syntax to work for some reason

Comment (tyleroooo) [reply src/views/menus/useGlobalCommands.tsx:85]: ah if you do `showVaults &&` you need to put a `.filter(isPresent)` at the end of the array declaration.

File: src/components/ComboboxMenu.tsx [hunk 2]
```diff
@@ -39,8 +39,8 @@ export const ComboboxMenu = <
   onItemSelected,
 
   title,
-  inputPlaceholder = 'Search…',
-  slotEmpty = 'No items found.',
+  inputPlaceholder = 'Search…', // TODO: i18n
```
Comment (moo-onthelawn) [inline src/components/ComboboxMenu.tsx:42]: oh yeah I don't think we can use `useStringGetter` in a default 🤔 eh the easiest way to get around this is to probably just use `inputPlaceholder ?? useStringGetter({key: STRING_KEYS.SEARCH})` in the code below (or if it's used in > 1 place, assign it somewhere at the top)

File: src/views/menus/useGlobalCommands.tsx [hunk 3]
```diff
@@ -154,5 +141,27 @@ export const useGlobalCommands = (): MenuConfig<string, string> => {
...
       })),
     },
+    {
+      // TODO: Remove this group when available under display settings
+      group: 'layout',
+      groupLabel: 'Layout', // TODO: i18n
+      items: [
+        {
+          value: LayoutItems.setDefaultLayout,
+          label: 'Set Default Layout',
```
Comment (moo-onthelawn) [inline src/views/menus/useGlobalCommands.tsx:151]: nit, I think these keys exist [here](https://github.com/dydxprotocol/v4-localization/blob/main/config/localization/en/app.json#L2007C8-L2007C26)! (though layout is still missing)

**Review by moo-onthelawn — APPROVED** [review]
🔥

File: src/views/menus/useGlobalCommands.tsx [hunk 4]
```diff
@@ -42,83 +46,62 @@ export const useGlobalCommands = (): MenuConfig<string, string> => {
...
-            dispatch(setAppColorMode(AppColorMode.RedUp));
-          },
+        showVaults && {
+          value: 'vaults',
+          slotBefore: <Icon iconName={IconName.Governance} />,
+          label: stringGetter({ key: STRING_KEYS.VAULTS }),
+          onSelect: () => navigate(AppRoute.Vaults),
         },
-      ],
+      ].filter((x) => !!x),
```
Comment (aforaleka) [inline src/views/menus/useGlobalCommands.tsx:82]: nit: I think you can actually use the function 'isPresent'

**Pull request merged**